plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -1496,13 +1500,29 @@ public VersionNumber getJenkinsVersion() {
      */
     @CheckForNull
     public VersionNumber getJenkinsVersionFromWar() {
```

Comment:
Hey @MarkEWaite 

Though I have implemented the solution, I am not sure how elegant it is.

**About testing:**

`PluginManagerIntegrationTest.failsWithMultiplePrerequisitesNotMet:206` fails. 

I did not understand the prerequisites to pass the test. 

I test using the following command with and without war file:
```
 java -jar D:/jenkins-plugin-manager-2.12.9.jar --plugin-file D:/plugins.txt --plugins delivery-pipeline-plugin:1.3.2 deployit-plugin
```

When the war file existed the command was successful. 
When the file was missing, this was the output:

```
Unable to open C:\ProgramData\Jenkins\jenkins.war
Unable to get Jenkins version from the WAR file C:\ProgramData\Jenkins\jenkins.war
War not found, installing all plugins: C:\ProgramData\Jenkins\jenkins.war
Done
```

How do I write a test case for this issue?

Reply:
I've added two tests in 5000bcd29ccdf2b9500d98ac9ddb91a3e85fbaef to provide an example of how you might test those methods.  I also fixed some logic errors that the tests exposed.  See 31c729c14dc49594d6d39c0c6e5cc49c531dad4e for those changes.

One or more additional tests are needed based on the coverage reporting.

Use the coverage reporting from `mvn -P enable-jacoco clean verify jacoco:report` to create a statement by statement coverage report so that you can see which parts of your new code are not being executed.  The HTML output of the report is written to the `target/site/jacoco` directory in each of the directories.

Reply:
Also, when you want a review, it is usually best to not have the pull request in a draft state.  Draft pull requests are usually interpreted as "Do not review this pull request".

Reply:
Thank you for reviewing. 

> One or more additional tests are needed based on the coverage reporting.

I will run the test coverage and get back.